crossdock/server/tch/gauntlet.go

```diff
@@ -0,0 +1,192 @@
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestNest(ctx thrift.Context, thing *gauntlet_apache.Xtruct2) (*gauntlet_apache.Xtruct2, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestMap(ctx thrift.Context, thing map[int32]int32) (map[int32]int32, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestStringMap(ctx thrift.Context, thing map[string]string) (map[string]string, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestSet(ctx thrift.Context, thing map[int32]bool) (map[int32]bool, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestList(ctx thrift.Context, thing []int32) ([]int32, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestEnum(ctx thrift.Context, thing gauntlet_apache.Numberz) (gauntlet_apache.Numberz, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestTypedef(ctx thrift.Context, thing gauntlet_apache.UserId) (gauntlet_apache.UserId, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	return thing, nil
+}
+
+func (thriftTestHandler) TestMapMap(ctx thrift.Context, hello int32) (map[int32]map[int32]int32, error) {
+	ctx.SetResponseHeaders(ctx.Headers())
+	result := map[int32]map[int32]int32{
+		-4: {
+			-4: -4,
+			-3: -3,
+			-2: -2,
+			-1: -1,
+		},
+		4: {
+			1: 1,
+			2: 2,
+			3: 3,
+			4: 4,
+		},
+	}
+	return result, nil
+}
+
+func (thriftTestHandler) TestInsanity(ctx thrift.Context, argument *gauntlet_apache.Insanity) (
+	map[gauntlet_apache.UserId]map[gauntlet_apache.Numberz]*gauntlet_apache.Insanity, error) {
+
+	ctx.SetResponseHeaders(ctx.Headers())
+	result := map[gauntlet_apache.UserId]map[gauntlet_apache.Numberz]*gauntlet_apache.Insanity{
+		1: {
+			gauntlet_apache.Numberz_TWO:   argument,
+			gauntlet_apache.Numberz_THREE: argument,
+		},
+		2: {
+			gauntlet_apache.Numberz_SIX: &gauntlet_apache.Insanity{},
+		},
+	}
+	return result, nil
+}
+
+func (thriftTestHandler) TestMulti(
+	ctx thrift.Context,
+	arg0 int8,
+	arg1 int32,
+	arg2 int64,
+	arg3 map[int16]string,
+	arg4 gauntlet_apache.Numberz,
+	arg5 gauntlet_apache.UserId,
+) (*gauntlet_apache.Xtruct, error) {
+
+	ctx.SetResponseHeaders(ctx.Headers())
+	hello := "Hello2"
+	result := &gauntlet_apache.Xtruct{
+		StringThing: &hello,
+		ByteThing:   &arg0,
+		I32Thing:    &arg1,
+		I64Thing:    &arg2,
+	}
+	return result, nil
+}
+
+func (thriftTestHandler) TestException(ctx thrift.Context, arg string) error {
+	ctx.SetResponseHeaders(ctx.Headers())
+	switch arg {
+	case "Xception":
+		code := int32(1001)
+		return &gauntlet_apache.Xception{ErrorCode: &code, Message: &arg}
+	case "TException":
+		// TODO is there something better I can raise here?
+		// unexpected exception.
+		return errors.New("great sadness")
```

Comment:
don't need to do anything special, any unknown error (e.g. not a thrift exception declared on this endpoint) will be handled as an unknown error in TChannel